### ISSUE TITLE
[expo-router] The action 'REPLACE' with payload {"name":"SCREEN"} was not handled by any navigator.

### ISSUE THREAD

**Issue by janusqa** [expo/expo#24353]
### Minimal reproducible example

https://github.com/janusqa/rn-bare/tree/bug_navigation

### Which package manager are you using? (Yarn is recommended)

npm

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

None

### Summary

On navigation from one nested stack to another nested stack (both nested in bottom tabs) gives the error
```
The action 'REPLACE' with payload {"name":"SCREEN"} was not handled by any navigator.
```

Download and run attached repo:
1) Click on  the second tab 
2) Click on the second link/button "Go to Tab2 / Screen 2"
3) Click on the first tab
4) Click on the second link "Go to Tab2 / Screen 1". 
5) Observe the error

**P.S. This is meant for expo-router. Not sure why it keeps ending up in main expo repo.  I begin by opening an issue in expo-router and after saving this is where its placed.**

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 5.4 Ubuntu 20.04.6 LTS (Focal Fossa)
      Shell: 5.0.17 - /bin/bash
    Binaries:
      Node: 18.14.0 - ~/.nvm/versions/node/v18.14.0/bin/node
      Yarn: 1.22.19 - ~/.nvm/versions/node/v18.14.0/bin/yarn
      npm: 9.4.2 - ~/.nvm/versions/node/v18.14.0/bin/npm
    SDKs:
      Android SDK:
        API Levels: 28, 29, 33, 34
        Build Tools: 30.0.3, 33.0.2, 34.0.0
        System Images: android-28 | Google Play Intel x86 Atom, android-29 | Google Play Intel x86 Atom, android-33 | Google APIs Intel x86_64 Atom, android-33 | Google Play Intel x86_64 Atom
    npmPackages:
      expo: ^49.0.8 => 49.0.8 
      react: 18.2.0 => 18.2.0 
      react-native: 0.72.4 => 0.72.4 
    Expo Workflow: managed
```

**Comment by devoren** [comment]
https://github.com/expo/router/issues/793

**Comment by marklawlor** [comment]
Fixed in `2.0.8`
